### PR TITLE
Bump main version to 9.0-alpha1

### PR DESCRIPTION
**Pull request by carlossanlop**
🎉

### CODE REVIEW

**Comment by carlossanlop** [comment]
@dotnet/ncl or @ViktorHofer do you know how to address the Quic failure message about opting into preview features?:

```
src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs(29,37): error CA2252: (NETCORE_ENGINEERING_TELEMETRY=Build) '_activeRequests''s type contains the preview type 'QuicStream' and requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2252)
```

**Comment by wfurt** [comment]
Strange, HTTP already has the preview feature set:

https://github.com/dotnet/runtime/blob/333c2c7eb28b1080aea379dab801778d3e992609/src/libraries/System.Net.Http/src/System.Net.Http.csproj#L6

Perhaps something in the SDK? 

BTW This may also break the Enterprise pipeline  - at least it always did in the past.

**Comment by jkoritzinsky** [comment]
EnablePreviewFeatures has an interesting behavior that it only works on the most current TFM. Looks like we might need some other solution here.

**Comment by wfurt** [comment]
removing `RequiresPreviewFeaturesAttribute` from `Quic` for now? I  would hope nobody would use 9.0 in production yet.

**Comment by jkoritzinsky** [comment]
Maybe manually including the attribute in the .NET 8 build (instead of using the SDK property) would work.

**Review by mmitche — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
@carlossanlop @wfurt I looked more into the failure. Here's whats happening:

- We have infrastructure to automatically say that for `<MajorVersion>N</MajorVersion>`, the latest TFM is `netN.0`.
- This PR updates `MajorVersion` to `9`.
- This PR does not update `NetCoreAppCurrent` to `net9.0`.
- As a result, System.Net.Http is building for `net8.0`.
- The SDK isn't applying the `EnablePreviewFeatures` attribute as `net8.0` is no longer the "latest TFM".

There are two possible solutions:

- Manually include the `EnablePreviewFeatures` attribute until we change `NetCoreAppCurrent`.
- Change `NetCoreAppCurrent` (and `NetCoreAppMinimum` and `NetCoreAppPrevious` as well) now and face the problems that come with that now.

@carlossanlop which would you rather do? I'd prefer the second option personally.

**Comment by jkoritzinsky** [comment]
I've started down the path of the second option locally. I'll try to finish it up tomorrow so I can push it out (just fixing up APICompat baselining).

File: src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml [hunk 1]
```diff
@@ -1,45 +1,46 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
   <Suppression>
     <DiagnosticId>CP0015</DiagnosticId>
     <Target>M:System.Diagnostics.Tracing.EventSource.Write``1(System.String,``0):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
-    <Left>ref/net8.0/System.Private.CoreLib.dll</Left>
-    <Right>lib/net8.0/System.Private.CoreLib.dll</Right>
+    <Left>ref/net9.0/System.Private.CoreLib.dll</Left>
+    <Right>lib/net9.0/System.Private.CoreLib.dll</Right>
```
Comment (ViktorHofer) [inline src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:8]: Feel free to remove all left and right entries from this file to avoid hardcoding the TFM. We usually only keep left and right when the suppression file applies to multiple assembly identities.

File: src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- Test renamed to TypeActivatorCreateInstanceUsesLongestAvailableConstructor. -->
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/CompatibilitySuppressions.xml:4]: undo

File: src/libraries/Microsoft.Extensions.DependencyInjection/src/CompatibilitySuppressions.xml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCodeAttribute was removed during .NET 8 with https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.DependencyInjection/src/CompatibilitySuppressions.xml:4]: undo

File: src/libraries/Microsoft.Extensions.Hosting/src/CompatibilitySuppressions.xml [hunk 4]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCode attributes removed during .NET 8: https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Hosting/src/CompatibilitySuppressions.xml:4]: undo

File: src/libraries/Microsoft.Extensions.Http/src/Microsoft.Extensions.Http.csproj [hunk 5]
```diff
@@ -33,7 +33,7 @@ System.Net.Http.IHttpClientFactory</PackageDescription>
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Options\src\Microsoft.Extensions.Options.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))">
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Http/src/Microsoft.Extensions.Http.csproj:36]: Please prefer `TFI == .NETCoreApp` when `NetCoreAppMinimum` is equal or greater than the min version in the condition. The MSBuild functions aren't for free.

File: src/libraries/Microsoft.Extensions.Logging/src/CompatibilitySuppressions.xml [hunk 6]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- RequiresDynamicCodeAttribute was removed during .NET 8 with https://github.com/dotnet/runtime/commit/f424c0f7ce8b1453d7124a3def8cf44a4ff42ead. -->
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Logging/src/CompatibilitySuppressions.xml:4]: undo

File: src/libraries/System.Security.Cryptography.Xml/src/CompatibilitySuppressions.xml [hunk 7]
```diff
@@ -3,7 +3,6 @@
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
   <Suppression>
     <DiagnosticId>CP0001</DiagnosticId>
-    <!-- Type must be public in implementation for serialization to work but we don't want to expose it publicly in the contract as it isn't public on .NET Framework. -->
```
Comment (ViktorHofer) [inline src/libraries/System.Security.Cryptography.Xml/src/CompatibilitySuppressions.xml:6]: undo

File: src/libraries/System.Reflection.Context/src/CompatibilitySuppressions.xml [hunk 8]
```diff
@@ -1,8 +1,22 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- The assembly version intentionally doesn't match between contract and implementation. -->
```
Comment (ViktorHofer) [inline src/libraries/System.Reflection.Context/src/CompatibilitySuppressions.xml:3]: undo

File: src/libraries/System.IO.Pipes/src/CompatibilitySuppressions.xml [hunk 9]
```diff
@@ -1,46 +1,88 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- Exposed public in System.IO.Pipes.AccessControl but implemented in System.IO.Pipes. -->
```
Comment (ViktorHofer) [inline src/libraries/System.IO.Pipes/src/CompatibilitySuppressions.xml:3]: undo

File: Directory.Build.targets [hunk 10]
```diff
@@ -49,6 +49,13 @@
     <SupportedNETCoreAppTargetFramework Include=".NETCoreApp,Version=v$(NETCoreAppMaximumVersion)" DisplayName=".NET $(NETCoreAppMaximumVersion)" Alias="net$(NETCoreAppMaximumVersion)" />
   </ItemGroup>
 
+  <!-- Add known packs from the downlevel TFM until we are building with an SDK that supports the new TFM all-up. -->
+  <ItemGroup Condition="'$(NetCoreAppToolCurrent)' != '$(NetCoreAppCurrent)'">
+    <KnownILLinkPack Include="@(KnownILLinkPack->WithMetadataValue('Identity', 'Microsoft.NET.ILLink.Tasks')->WithMetadataValue('TargetFramework', '$(NetCoreAppToolCurrent)'))">
+      <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    </KnownILLinkPack>
+  </ItemGroup>
```
Comment (ViktorHofer) [inline Directory.Build.targets:57]: Please put this into targetingpacks.targets

**Comment by lewing** [comment]
```
.dotnet/sdk/8.0.100-rc.1.23415.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1211: (NETCORE_ENGINEERING_TELEMETRY=Restore) EnableSingleFileAnalyzer is not supported for the target framework. Consider multi-targeting to a supported framework to enable single-file analysis, and set EnableSingleFileAnalyzer only for the supported frameworks. For example:
<EnableSingleFileAnalyzer Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableSingleFileAnalyzer>
```

File: src/coreclr/nativeaot/System.Private.CoreLib/src/CompatibilitySuppressions.xml [hunk 11]
```diff
@@ -959,43 +959,43 @@
   <Suppression>
     <DiagnosticId>CP0015</DiagnosticId>
     <Target>M:System.Diagnostics.Tracing.EventSource.Write``1(System.String,``0):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
-    <Left>ref/net8.0/System.Private.CoreLib.dll</Left>
-    <Right>lib/net8.0/System.Private.CoreLib.dll</Right>
+    <Left>ref/net9.0/System.Private.CoreLib.dll</Left>
+    <Right>lib/net9.0/System.Private.CoreLib.dll</Right>
```
Comment (ViktorHofer) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/CompatibilitySuppressions.xml:963]: Remove these lefts and rights from this file as well.

File: src/coreclr/crossgen-corelib.proj [hunk 12]
```diff
@@ -158,4 +158,6 @@
     <Message Importance="High" Text="Crossgenning of System.Private.CoreLib succeeded." />
     <Message Importance="High" Text="Product binaries are available at $(BinDir)" />
   </Target>
+
+  <Import Project="$(RepositoryEngineeringDir)targetingpacks.targets" />
```
Comment (ViktorHofer) [inline src/coreclr/crossgen-corelib.proj:162]: Why is that import needed in a NoTargets SDK project?

File: src/coreclr/runtime.proj [hunk 13]
```diff
@@ -6,6 +6,7 @@
 
     <_BuildNativeTargetOS>$(TargetOS)</_BuildNativeTargetOS>
     <_BuildNativeTargetOS Condition="'$(TargetsLinuxBionic)' == 'true'">linux-bionic</_BuildNativeTargetOS>
+    <RunAnalyzers>false</RunAnalyzers>
```
Comment (ViktorHofer) [inline src/coreclr/runtime.proj:9]: Why is this necessary? It would be good to set RunAnalyzers to false globally for NoTargets SDK projects. See https://github.com/dotnet/runtime/blob/296a1d5b9fcea725d3ba7c7b5991885a8f99dbd2/Directory.Build.props#L410

File: src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj [hunk 14]
```diff
@@ -9,12 +9,16 @@
...
+    -->
+    <IsTrimmable>false</IsTrimmable>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" />
     <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
   </ItemGroup>
-
```
Comment (ViktorHofer) [inline src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj:19]: Please undo this deletion. We aim towards having our project files represent the `dotnet new` ones which include a new line after the project start tag and before the project close tag.

File: src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj [hunk 15]
```diff
@@ -9,12 +9,16 @@
     <OutputPath>$(RuntimeBinDir)/ilc-published/netstandard</OutputPath>
     <Configurations>Debug;Release;Checked</Configurations>
     <PlatformTarget>AnyCPU</PlatformTarget>
+    <!--
+      The SDK warns if a project targets a framework that doesn't support trimming and IsTrimmable is true.
+      We set IsTrimmable higher up in the Directory.Build.props files, so we reset it to false here.
+    -->
+    <IsTrimmable>false</IsTrimmable>
```
Comment (ViktorHofer) [inline src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj:16]: So netstandard2.0 doesn't support trimming, at all? If so, should we set this globally in a targets file based on the TFM?

**Comment by ManickaP** [comment]
What about just `<NoWarn>CA2252</NoWarn>`?

File: src/coreclr/runtime.proj (hunk 13 above)
Comment (jkoritzinsky) [reply src/coreclr/runtime.proj:9]: We enable some of the "Form Factor" analyzers like the Trim analyzer and the SingleFile analyzer globally in the repo. Now the SDK warns if you try to enable any of these analyzers and you don't have an ILLink pack available for the target TFM. As a result, it's tripping on all of our NoTargets projects. I'll move this check up to there.

File: src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj (hunk 15 above)
Comment (jkoritzinsky) [reply src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj:16]: Yes, we don't have trimming support for .NET Standard 2.0 (it's completely unannotated, so we can't have accurate analysis).

Comment (ViktorHofer) [reply src/coreclr/tools/aot/ILCompiler.Build.Tasks/ILCompiler.Build.Tasks.csproj]: Then shouldn't we disable this globally in a targets file for all projects?

File: src/coreclr/tools/aot/Directory.Build.targets [hunk 16]
```diff
@@ -1,6 +1,6 @@
 <Project>
-  <Import Project="../Directory.Build.props" />
   <PropertyGroup>
     <IsTrimmable Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</IsTrimmable>
   </PropertyGroup>
+  <Import Project="../Directory.Build.targets" />
```
Comment (ViktorHofer) [inline src/coreclr/tools/aot/Directory.Build.targets:5]: Please move the import above the custom property as we do in other Directory.Build.targets files: https://github.com/dotnet/runtime/blob/64243bbf5e9ee53c0c4c5678f2cd8c7f1c9b4f6f/src/libraries/System.Runtime.Caching/Directory.Build.targets#L2

File: src/installer/pkg/sfx/Directory.Build.props [hunk 17]
```diff
@@ -10,6 +10,8 @@
     <ProductBrandPrefix>Microsoft .NET</ProductBrandPrefix>
     <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
     <UseRuntimePackageDisclaimer>true</UseRuntimePackageDisclaimer>
+    <!-- We don't have any sources here, so don't run analyzers. -->
+    <RunAnalyzers>false</RunAnalyzers>
```
Comment (ViktorHofer) [inline src/installer/pkg/sfx/Directory.Build.props:14]: You could also condition on shared framework packages with the following property: https://github.com/dotnet/runtime/blob/64243bbf5e9ee53c0c4c5678f2cd8c7f1c9b4f6f/Directory.Build.props#L408

File: Directory.Build.props [hunk 18]
```diff
@@ -429,6 +429,11 @@
     <PublishWindowsPdb>false</PublishWindowsPdb>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true'">
+    <!-- Explicitly disable running analyzers to avoid trying to discover the correct ILLink tool pack for a NoTargets project. -->
+    <RunAnalyzers>false</RunAnalyzers>
+  </PropertyGroup>
```
Comment (ViktorHofer) [inline Directory.Build.props:435]: nit: can you please move this into Analyzers.targets? We already have some conditions there to determine when to not run analyzers.

File: src/installer/pkg/sfx/Directory.Build.props (hunk 17 above)
Comment (jkoritzinsky) [reply src/installer/pkg/sfx/Directory.Build.props:14]: I can't as that property isn't set early enough to be available in Directory.Build.props files. I tried it and it didn't work.

File: Directory.Build.props [hunk 19]
```diff
@@ -80,20 +80,20 @@
 
   <!-- The TFMs to build and test against. -->
   <PropertyGroup>
-    <NetCoreAppCurrentVersion>8.0</NetCoreAppCurrentVersion>
+    <NetCoreAppCurrentVersion>9.0</NetCoreAppCurrentVersion>
```
Comment (carlossanlop) [inline Directory.Build.props:83]: This wasn't supposed to happen in this PR, right? @ViktorHofer

Comment (jkoritzinsky) [reply Directory.Build.props:83]: Basically we had the option to either work around the fact that we weren't doing this with the EnablePreviewFeatures work or just doing it and getting it over with.

I decided to do it as I didn't see a reason to work around the EnablePreviewFeatures issue instead of just fixing it and I thought it would be a much easier change than it ended up being.

If we want this to be in a separate branch, I can branch off from here with those changes in a draft PR, revert them here, and do the `EnablePreviewFeatures` workaround.

Comment (lewing) [reply Directory.Build.props:83]: It is unlikely that we can change the target here in a single pass, the sdk typically takes months to adjust

Comment (jkoritzinsky) [reply Directory.Build.props:83]: I think we're actually pretty close to updating the TFM. Looks like we need to update the bootstrapping scripts for the installer tests and a little bit of debugging of the WBT legs.

Next week I'll branch off the upgrading to another PR and do the quick workaround for the EnablePreviewFeatures attribute to unblock this PR.

File: Directory.Build.props [hunk 20]
```diff
@@ -80,20 +80,20 @@
...
 
     <!-- The previous supported .NET version. -->
-    <NetCoreAppPreviousVersion>7.0</NetCoreAppPreviousVersion>
+    <NetCoreAppPreviousVersion>8.0</NetCoreAppPreviousVersion>
     <NetCoreAppPrevious>net$(NetCoreAppPreviousVersion)</NetCoreAppPrevious>
     <NetCoreAppPrevious Condition="'$(DotNetBuildFromSource)' == 'true'">$(NetCoreAppCurrent)</NetCoreAppPrevious>
 
     <!-- The minimum supported .NET version. -->
-    <NetCoreAppMinimum>net6.0</NetCoreAppMinimum>
+    <NetCoreAppMinimum>net8.0</NetCoreAppMinimum>
```
Comment (ericstj) [inline Directory.Build.props:96]: We still support net6.0 for another year - are we really ready to move this?  Also I expect we will have projects that expect Previous and Minimum to be different.  @ViktorHofer

Comment (jkoritzinsky) [reply Directory.Build.props:96]: Doesn't .NET 6.0 go out of support on the day that .NET 9 comes out?

File: src/installer/pkg/sfx/Directory.Build.props (hunk 17 above)
Comment (ViktorHofer) [reply src/installer/pkg/sfx/Directory.Build.props]: Consider setting it in analyzers.targets. At that point, the property should already be set.

File: Directory.Build.props (hunk 20 above)
Comment (ViktorHofer) [reply Directory.Build.props:96]: We did the same when we switched from .NET 6 to .NET 7: We removed `netcoreapp3.1` and `net5.0` support as when .NET 7 released, those runtimes were already officially out-of-support.

That model worked well with .NET 7 and I would prefer to drop the TFMs now than in the middle of .NET 9 development.

> Also I expect we will have projects that expect Previous and Minimum to be different.

We should fix such wrong assumptions. I don't recall any managed project that requires those two properties to differ. Those were designed to be equal every odd major release version. And, source build already sets all three properties to the same value.

File: eng/Analyzers.targets [hunk 21]
```diff
@@ -1,4 +1,8 @@
 <Project>
+  <PropertyGroup Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true' or '$(UsingMicrosoftDotNetSharedFrameworkSdk)' == 'true'">
+    <!-- Explicitly disable running analyzers to avoid trying to discover the correct ILLink tool pack for a project that has no sources. -->
+    <RunAnalyzers>false</RunAnalyzers>
+  </PropertyGroup>
```
Comment (ViktorHofer) [inline eng/Analyzers.targets]: nit: Personally I would prefer this to be a condition directly on the property instead of defining a new group but that's just my own taste of style. We did the same for the DotNetBuildFromSource condition below...

**Comment by ViktorHofer** [comment]
To fix the installer failures, you need to update https://github.com/dotnet/runtime/blob/9ddb7316837cafecceca637dabb1a9b1f2b08409/src/installer/tests/Assets/TestUtils/TestProjects.targets#L7-L31

**Comment by radical** [comment]
I'm looking at the wasm failures now.

**Comment by jkoritzinsky** [comment]
I've reverted the TFM upgrade to avoid bundling together too many different concepts in this PR, as requested.

I'll open a new draft PR in a minute with the TFM changes. @radical I'll revert your commit here and cherry-pick it to the other PR.

**Comment by jkoritzinsky** [comment]
PR with the TFM upgrade is at https://github.com/dotnet/runtime/pull/90880

**Comment by carlossanlop** [comment]
> PR with the TFM upgrade is at #90880

Thank you so much, @jkoritzinsky, that is a huge contribution. I think it's best to send it separately.

**Comment by jkoritzinsky** [comment]
> > PR with the TFM upgrade is at #90880
> 
> Thank you so much, @jkoritzinsky, that is a huge contribution. I think it's best to send it separately.

I agree. This PR only has the changes required for the rebranding, not the TFM upgrade.

File: src/libraries/System.Linq.Expressions/src/CompatibilitySuppressions.xml [hunk 22]
```diff
@@ -139,6 +139,48 @@
     <Left>ref/net8.0/System.Linq.Expressions.dll</Left>
     <Right>lib/net8.0/System.Linq.Expressions.dll</Right>
   </Suppression>
+  <Suppression>
```
Comment (ViktorHofer) [inline src/libraries/System.Linq.Expressions/src/CompatibilitySuppressions.xml:142]: These changes should be reverted

Comment (jkoritzinsky) [reply src/libraries/System.Linq.Expressions/src/CompatibilitySuppressions.xml:142]: The build fails without these two changes.

**Comment by radical** [comment]
New sdk error on windows:
```
EXEC : error : dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have [D:\a\_work\1\s\src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj]
##[error]EXEC(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have
   enough disk space.
  dotnet-install: Extracting the archive.
```

The log doesn't show the exit code. I'll check locally.

**Comment by radical** [comment]
> New sdk error on windows:
> 
> ```
> EXEC : error : dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have [D:\a\_work\1\s\src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj]
> ##[error]EXEC(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) dotnet-install: Failed to check the disk space. Installation will continue, but it may fail if you do not have
>    enough disk space.
>   dotnet-install: Extracting the archive.
> ```
> 
> The log doesn't show the exit code. I'll check locally.

I'm not sure what is wrong here. The `dotnet-install.ps1` script uses `Write-Warning` to emit the message, and this script is being run with `Exec` msbuild task with `IgnoreStandardErrorWarningFormat="true"`. I don't know how it is becoming an error. The task doesn't even seem to fail the build, and instead the whole build completes, and then msbuild shows the error at the end.

**Review by ViktorHofer — COMMENTED** [review]
The AssemblyVersion property in Versions.props must be hardcoded to 8.0.0.0 for now. We had the same issue last release and noticed that updating the TFM must happen in conjunction with reving the major assembly version.

Please also undo the changes in both CompatibilitySuppressions.xml, the TestData.resources one and the PACKAGE.md change which looks unrelated.

**Review by ViktorHofer — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
@radical if the WASM MT failures are known, can you merge this in?
